### ISSUE TITLE
repo stats

### ISSUE THREAD

**Issue by prettymuchbryce** [ipfs-shipyard/ipfs-webui#629]
Is there any interest in some UI which displays how much space is remaining in the repo? I currently don't see this information available from anywhere within the UI.

We could use the `/api/v0/stats/repo` api to display some sort of bar for `RepoSize` / `StorageMax`.

**Comment by lidel** [comment]
Yup, actually, it is currently being designed! Status widget will be initially introduced in IPFS Desktop, but will also land in WebUI (when we get back to it). 

See initial mockup at: https://github.com/ipfs-shipyard/pm-ipfs-gui/issues/12#issuecomment-361733414  (last one)

**Comment by olizilla** [comment]
It'd be great to show the repo stats. 

A good solution for this will need to take into account that the `StorageMax` value is a guide that indicates when ipfs should run GC on the repo to try and free some space. `StorageMax` is not a hard limit and if GC isn't enabled, it's entirely possible the reported `RepoSize` could exceed the `StorageMax` value, so we should visualise it in a way that shows that clearly with `StorageMax` as a threshold rather than a max value.

> `StorageMax` A soft upper limit for the size of the ipfs repository's datastore. With StorageGCWatermark, is used to calculate whether to trigger a gc run (only if --enable-gc flag is set).

https://github.com/ipfs/go-ipfs/blob/1e0d53fe3cb32be903d768495fd0526002a03c63/docs/config.md#datastore

**Comment by olizilla** [comment]
Comment from @lidel in https://github.com/ipfs-shipyard/ipfs-webui/issues/887#issuecomment-440330288

---
 Here's an idea: replace CountryChart with a "StorageChart" – a pie chart composed from various sources:
-  `RepoSize` vs `StorageMax` from `ipfs stats repo --size-only` 
-  `CumulativeSize('/')` from `ipfs files stat / --size`  (and maybe lazy-loaded `ipfs files stat / --size --with-local` which is expensive)

How is this useful?
- If GC is ON: You see actual size of your repo, available space etc. 
   - Each segment could be clickable (`cursor: help`) and open docs explaining IPFS concepts  (opportunity to improve docs, as we don't have "ipfs repository" on the [concepts](https://docs.ipfs.io/guides/concepts/) page yet)
   - Could be a good opportunity to identify missing APIs to make it even more useful. 
- If  GC is OFF: People may have GC disabled but not know it. We could compare (`StorageMax`&`RepoSize`) and if `RepoSize` is bigger, use it as 100% and display the chunk over `StorageMax` in red. Clicking on it or hovering could display information about GC (how it works, how to enable it).

I would like to hear your thoughts on this.

**Comment by andrasfuchs** [comment]
@olizilla I agree, your suggestion would be more practical than the one we have now on the Status page.

I'm a newbie to IPFS, so I don't know if the following stats are available in IPFS or even possible to get, but I think it would be great to show these somewhere (probably on the Files page):
* How much of my content (the files/folders pinned by me) was downloaded from my node
* How many times were those files/folders pinned by others
* How many of those others who have the files (the other seeders) are currently online
* How many times were my IPNS entries accessed

**Comment by jessicaschilling** [comment]
Related: https://github.com/ipfs/ipfs-gui/issues/39 -- adding controls for node start/stop, bandwidth/size management.
